### PR TITLE
trivial: Move some EFI variable setup code to libfwupdplugin

### PR DESCRIPTION
**Pull request by hughsie**
This allows us to share the 'create a plausible UEFI system' code to be used in the other self test binaries.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Review by superm1 — DISMISSED** [review]
one small suggestion (feel free to ignore).

File: libfwupdplugin/fu-efivars.c [hunk 1]
```diff
@@ -403,6 +408,23 @@ fu_efivars_secure_boot_enabled(FuEfivars *self, GError **error)
 	return FALSE;
 }
 
+/**
+ * fu_efivars_set_secure_boot: (skip):
+ **/
+gboolean
+fu_efivars_set_secure_boot(FuEfivars *self, guint8 value, GError **error)
```
[the remark this answers is not in the document]
Comment (hughsie) [reply libfwupdplugin/fu-efivars.c]: I think that makes sense too. Fixed!

File: plugins/uefi-capsule/fu-uefi-capsule-plugin.c [hunk 2]
```diff
@@ -926,10 +929,12 @@ static void
 fu_uefi_capsule_plugin_test_secure_boot(FuPlugin *plugin)
 {
 	FuEfivars *efivars = fu_context_get_efivars(fu_plugin_get_context(plugin));
-	const gchar *result_str = "Disabled";
-	if (fu_efivars_secure_boot_enabled(efivars, NULL))
-		result_str = "Enabled";
-	fu_plugin_add_report_metadata(plugin, "SecureBoot", result_str);
+	gboolean secureboot_enabled = FALSE;
+
+	fu_efivars_get_secure_boot(efivars, &secureboot_enabled, NULL);
```
Comment (superm1) [inline plugins/uefi-capsule/fu-uefi-capsule-plugin.c:934]: I don't think it's worth ignoring the error here, at least log it

Comment (hughsie) [reply plugins/uefi-capsule/fu-uefi-capsule-plugin.c:934]: maybe set the report attr to `Unknown`? That'd be an interesting datapoint.

Comment (superm1) [reply plugins/uefi-capsule/fu-uefi-capsule-plugin.c:934]: Yeah; that's what I was thinking - we should know if there is such an error here ever.

Comment (hughsie) [reply plugins/uefi-capsule/fu-uefi-capsule-plugin.c]: I put the error message there -- that way we can tell the difference between an empty var and a missing var.

**Review by superm1 — APPROVED** [review]

**Pull request merged**